activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -894,6 +894,10 @@ def remove_index_for_alter(table_name, column_name = nil, **options)
           "DROP INDEX #{quote_column_name(index_name)}"
         end
 
+        def supports_values_syntax?
+          !mariadb? && database_version >= "8.0.19"
```

Comment:
I wonder if there is a way to detect vitess like we do MySQL / MariaDB?

Reply:
I cannot check myself what vitess returns in the version string, but I'm going to ask around to see if there is a possibility to do smth like that

Reply:
Yeah, no rush, I'll merge as is, and if you find a way to detect it, please open a followup.

Reply:
@maxprokopiev the default full version strings for vitess will look like:
* v15 and earlier (end of life): `5.7.9-vitess-<vitess_version>` e.g `5.7.9-vitess-15.0.5`
* v16 and later: `8.0.30-Vitess`

So a case insensitive match on `vitess` would likely be sufficient. Vitess users can override the version string that it returns, but as long as they include `vitess` that would work.